### PR TITLE
SALTO-2263 add netsuite client validation

### PR DESCRIPTION
**Pull request by dantal4**
add NetSuite client validation

---
_Release Notes_: 
Netsuite Adapter:
- add client validation

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/client/sdf_client.ts [hunk 1]
```diff
@@ -392,7 +396,7 @@ export default class SdfClient {
 
   private async initProject(suiteAppId?: string): Promise<Project> {
     const authId = uuidv4()
-    const projectName = `TempSdfProject-${authId}`
+    const projectName = `sdf-${authId}`
```
Comment (alonstern) [inline packages/netsuite-adapter/src/client/sdf_client.ts:399]: why?

File: packages/netsuite-adapter/src/client/sdf_client.ts [hunk 2]
```diff
@@ -936,25 +940,74 @@ export default class SdfClient {
     )
   }
 
+  private static getValidationErrorMessage(
+    errorMessageLines: string[],
+    filePath: string
+  ): string | undefined {
+    const invalidObjectErrorMessage = `Errors for file ${filePath}.`
```
Comment (alonstern) [inline packages/netsuite-adapter/src/client/sdf_client.ts:947]: performance nit: You go over all the output for each file. Can you go over it once and return a map for filePath to error?

File: packages/netsuite-adapter/src/change_validator.ts [hunk 3]
```diff
@@ -80,6 +101,12 @@ const getChangeValidator: ({ withSuiteApp, warnStaleData, fetchByQuery, deployRe
       const changeErrors: ChangeError[] = _.flatten(await Promise.all([
         ...validators.map(validator => validator(changes)),
         warnStaleData ? safeDeployValidator(changes, fetchByQuery, deployReferencedElements) : [],
+        validate ? netsuiteClientValidation(
```
[the remark this answers is not in the document]
Comment (dantal4) [reply packages/netsuite-adapter/src/change_validator.ts:104]: the change validator runs `NetsuiteClient.validate`, which in the future can include more "active" validations other than `sdfValidation`

File: packages/netsuite-adapter/src/client/sdf_client.ts (hunk 1 above)
Comment (dantal4) [reply packages/netsuite-adapter/src/client/sdf_client.ts:399]: the validation is more strict in some things than the deploy itself, one of them is a maximum of 40 chars in the project name.

**Review by alonstern — APPROVED** [review]
LGTM 👌

**Pull request merged**